autoroutes.pyx

```diff
@@ -47,16 +65,51 @@ cdef class Edge:
             str rest = self.pattern[prefix_len:]
         new_child.connect(self.child, rest)
         self.child = new_child
+        new_child.compile()
         self.pattern = self.pattern[:prefix_len]
 
+    cdef Node join(self, str path):
+        cdef:
+            unsigned int local_index, candidate_index
+            Edge candidate = Edge(path, None)
+        candidate.compile()
```

Comment:
Add a comment explaining why the compilation is required before the compilation, maybe do that within the `compare`?

Reply:
I'm thinking about calling it in the `__init__.py` once and for all.